relay.d.ts

```diff
@@ -0,0 +1,43 @@
+export declare const re: {
+  BINARY_BROADCAST: Uint8Array,
+  RELAY_QUERY: Uint8Array,
+  TEXT_BROADCAST: string,
+  BINARY: number,
+  TEXT: number,
+  OP: {
+    ANNOUNCE: number,
+    ANNOUNCE_REPLY: number,
+  },
+  USERS: {},
+  UInt8UserIdToBase64: (userId: Uint8Array) => string,
+  Base64ToUInt8UserID: (userId: string) => Uint8Array,
+  ArrayToStr: (x: array) => string,
+  StrToArray: (x: string) => array,
+  UserIdHex: (userId: Uint8Array) => string,
+};
+
+export declare const ArrayEq: (a: array, b: array) => boolean;
+export declare const UInt8UserIdToBase64: (userId: Uint8Array) => string;
+export declare const Base64ToUInt8UserID: (userId: string) => Uint8Array;
+
+export declare class Relay {
+  public ws: WebSocket;
+  public onClientDisconnect: ((userId: Uint8Array) => void) | null;
+  public onSubscription: ((userId: Uint8Array) => Relay) | null;
+  public BinaryMessageHandlers: { [key: keyof re.OP]: (sender: Uint8Array, message: Uint8Array | object) => void };
```

Comment:
The key here is really just open-ended. I don't think you want to require it to be part of re.OP, it's just a number and an application could define its own opcode list.

Reply:
I should give more context, re.OP is just an example for how opcodes should be used with the library but it isn't really a required part of the library.  It could be entirely removed or maybe refactored into an example application instead.  The opcode is totally application specific.